lints/lint_extra_subject_common_names.go

```diff
@@ -0,0 +1,51 @@
+/*
+ * ZLint Copyright 2019 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package lints
+
+import (
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+)
+
+type extraSubjectCommonNames struct{}
+
+func (l *extraSubjectCommonNames) Initialize() error {
+	return nil
+}
+
+func (l *extraSubjectCommonNames) CheckApplies(c *x509.Certificate) bool {
+	return util.IsSubscriberCert(c)
+}
+
+func (l *extraSubjectCommonNames) Execute(c *x509.Certificate) *LintResult {
+	// Multiple subject commonName fields are not expressly prohibited by section
+	// 7.1.4.2.2 but do seem to run afoul of the intent. For that reason we return
+	// only a Notice level finding here.
+	if len(c.Subject.CommonNames) > 1 {
+		return &LintResult{Status: Notice}
```

Comment:
I think this should just be a warning rather than a notice. The lint is named that way already, and I don't think there's much contention that it's against community best practices to have more than one CN in a certificate.

Reply:
Hrm, seems to violate what we have in our README and how we designed levels originally. This certainly seems more than just an "INFO" though. I'm not sure what we do about community standards. I suspect that people just ignore `INFO`s. This might be best thought about similar to the Mozilla lints where a source might just be "Community Best Practices" that aren't as stringent as BR or RFC set. Thoughts?

Regardless, name of lint should match what it returns.

Reply:
> Hrm, seems to violate what we have in our README and how we designed levels originally. This certainly seems more than just an "INFO" though. 

Agreed. Personally for this case I'm happy to use Warn even if it doesn't violate a direct SHOULD/SHOULD-NOT. I think what's in the README should be considered advice and not law.

Do you feel OK calling it a Warn on the basis of two votes or should we try to solicit input from some of the other people using zlint? 

>  I suspect that people just ignore INFOs.

I suspect so too.

> Regardless, name of lint should match what it returns.

+1. I've put it to Warn to match the name in branch. If consensus goes another way I'll switch the name.

Reply:
@tadukurow @cardonator Do you folks have a preference for warn vs info for this lint?

Reply:
SGTM. I forgot it's American thanksgiving and the folks I tagged might not be around to give input.

I'm going to merge this now. If someone has strong feelings after the fact there will be time between the merge & cutting a 1.1.0 release tag where we can land an adjustment. I'll aim to cut that tag later next week.

Reply:
@cpu in our case, I would say WARN is fine. I agree that even though it's not codified in the BRs, it is heavily frowned upon in CABF and SCWG. Also with the recent discussion of "default-deny" in the lists it just makes sense to lint for something that would cause a commotion.

Reply:
@cardonator Thanks for weighing in. Appreciated!